dashboard/modules/job/tests/test_http_job_server.py

```diff
@@ -98,6 +109,96 @@ def test_submit_job(job_sdk_client, working_dir_option):
     assert stderr == working_dir_option["expected_stderr"]
 
 
+def test_http_bad_request(job_sdk_client):
+    """
+    Send bad requests to job http server and ensure right return code and
+    error message is returned via http.
+    """
+    client = job_sdk_client
+
+    # 400 - HTTPBadRequest
+    with pytest.raises(requests.exceptions.HTTPError) as e:
+        _ = client._do_request(
+            "POST",
+            JOBS_API_ROUTE_SUBMIT,
+            json_data={"key": "baaaad request"},
+        )
+
+    ex_message = str(e.value)
+    assert "400 Client Error" in ex_message
+    assert "TypeError: __init__() got an unexpected keyword argument" in ex_message  # noqa: E501
+
+    # 405 - HTTPMethodNotAllowed
+    with pytest.raises(requests.exceptions.HTTPError) as e:
+        _ = client._do_request(
+            "GET",
+            JOBS_API_ROUTE_SUBMIT,
+            json_data={"key": "baaaad request"},
+        )
+    ex_message = str(e.value)
+    assert "405 Client Error: Method Not Allowed" in ex_message
+
+    # 500 - HTTPInternalServerError
+    with pytest.raises(requests.exceptions.HTTPError) as e:
+        _ = client.submit_job(
+            entrypoint="echo hello",
+            runtime_env={"working_dir": "s3://does_not_exist"})
+    ex_message = str(e.value)
+    assert "500 Server Error" in ex_message
```

Comment:
I think this should be a 400, not a 500

Reply:
(500 means something went wrong, like an internal assertion failed or unexpected error occurred)

Reply:
i wrapped everything within `job_manager.submit_job()` in 500 to wrap everything happens at job manager / ray level, for this one in particular it fails at `validation.py` for not having a .zip suffix for s3 path, maybe we can add a few more cases and see. Probably not all of them are 500